test/e2e/serial/tests/operator_profile_update.go

```diff
@@ -0,0 +1,282 @@
+				dsObj := appsv1.DaemonSet{}
+				err = fxt.Client.Get(context.TODO(), client.ObjectKey(dsKey), &dsObj)
+				Expect(err).ToNot(HaveOccurred())
+				cnt := k8swgobjupdate.FindContainerByName(dsObj.Spec.Template.Spec.Containers, manifests.ContainerNameRTE)
+				Expect(cnt).NotTo(BeNil(), "cannot find container data for %q", manifests.ContainerNameRTE)
+				return cnt.Args
+
+			}).WithTimeout(time.Minute).WithPolling(9*time.Second).Should(ContainElement(ContainSubstring("--no-publish")), "ds was not updated as expected: \"--no-pubish\" arg is missing.")
+			klog.Info("waiting for DaemonSet to be ready")
+			_, err = wait.With(e2eclient.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(context.TODO(), dsKey)
+			Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset %s: %v", dsKey.String(), err)
+		})
+
+		AfterEach(func() {
+			restoreNROP(fxt, *nropObjInitial)
+		})
+
+		It("[tier1] should make a best-effort pod running", func() {
+			By("create best-effort pod expect it to start running")
+			testPod := objects.NewTestPodPause(fxt.Namespace.Name, "testpod")
+			testPod.Spec.SchedulerName = serialconfig.Config.SchedulerName
+			err := fxt.Client.Create(context.TODO(), testPod)
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedPod, err := wait.With(fxt.Client).Timeout(5*time.Minute).ForPodPhase(context.TODO(), testPod.Namespace, testPod.Name, corev1.PodRunning)
+			if err != nil {
+				_ = objects.LogEventsForPod(fxt.K8sClient, updatedPod.Namespace, updatedPod.Name)
+			}
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("[tier1] should make a burstable pod pending", func() {
+			By("create burstable pod and expect it to keep pending")
+			testPod = objects.NewTestPodPause(fxt.Namespace.Name, "testpod")
+			testPod.Spec.SchedulerName = serialconfig.Config.SchedulerName
+			testPod.Spec.Containers[0].Resources.Requests = corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("256Mi"),
+			}
+			err := fxt.Client.Create(context.TODO(), testPod)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = wait.With(fxt.Client).Interval(10*time.Second).Steps(3).WhileInPodPhase(context.TODO(), testPod.Namespace, testPod.Name, corev1.PodPending)
+			if err != nil {
+				_ = objects.LogEventsForPod(fxt.K8sClient, testPod.Namespace, testPod.Name)
+			}
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("verify pod events contain the correct message %q", eventMsgInvalidTopology))
+			ok, err := verifyPodEvents(fxt, testPod, eventReasonFailedScheduling, eventMsgInvalidTopology)
+			Expect(err).ToNot(HaveOccurred())
+			if !ok {
+				_ = objects.LogEventsForPod(fxt.K8sClient, testPod.Namespace, testPod.Name)
+			}
+			Expect(ok).To(BeTrue(), "failed to find the expected event with Reason=%q and Message contains %q", eventReasonFailedScheduling, eventMsgInvalidTopology)
+
+			By("restore NROP configuration and verify the pod goes running")
+			restoreNROP(fxt, *nropObjInitial)
```

Comment:
so we can try to restore twice?

Reply:
right, but we do check if a restore is needed before actually applying the initial config (again).